<launch>
    <!-- Node Settings -->
    <arg name="output" default="screen"/>
    <!-- Verbose logging -->
    <arg name="verbose" default="false"/>
    <!-- Frame ID for me1sages -->
    <arg name="frame_id" default="ukf"/>

<group ns = "ukf_ns">
    <node pkg="ukf_test" name="ukf_node" type="ukf_test_node" output="$(arg output)">
        <param name="imu_topic" type="string" value="/imu_ns/imu/imu" />
        <param name="frame_id" type="string" value="$(arg frame_id)"/>
        <param name="imu_rate" type="int" value="1000" />
        <param name="imu_delay" type="int" value="30" />
        <param name="predict_rate" type="int" value="150" />
        <param name="update_part/vio_topic" type="string" value="/fla/vio/odom"/>
        <param name="update_part/alpha"  value="0.5" />
        <param name="update_part/beta"  value="2.0"/>
        <param name="update_part/kappa"  value="-12.0"/>
        <param name="update_part/verbose"  value="$(arg verbose)"/>
        <param name="update_part/vm_con_P"  value="0.01"/>
        <param name="update_part/vm_con_V"  value="0.001"/>
        <param name="update_part/vm_con_Att"  value="0.1"/>
        <param name="update_part/sm_con_P"  value="0.01"/>
        <param name="update_part/sm_con_V"  value="0.001"/>
        <param name="update_part/sm_con_Att"  value="0.1"/>
        <param name="update_part/sm_con_ba"  value="1e-6"/>
        <param name="update_part/sm_con_bw"  value="1e-6"/>
        <param name="update_part/sm_con_na"  value="2e-7"/>
        <param name="update_part/sm_con_nw"  value="2e-7"/>
    </node>
</group>
</launch>